src/txn/txn_timestamp.c

```diff
@@ -302,6 +302,32 @@ __txn_global_query_timestamp(
 done:	__wt_timestamp_set(tsp, &ts);
 	return (0);
 }
+
+/*
+ * __txn_query_timestamp --
+ *	Query a timestamp.
+ */
+static int
+__txn_query_timestamp(
+    WT_SESSION_IMPL *session, wt_timestamp_t *tsp, const char *cfg[])
+{
+	WT_CONFIG_ITEM cval;
+	WT_TXN *txn;
+	wt_timestamp_t ts;
+
+	txn = &session->txn;
+
+	WT_STAT_CONN_INCR(session, session_query_ts);
+	WT_RET(__wt_config_gets(session, cfg, "get", &cval));
+	if (WT_STRING_MATCH("read_timestamp", cval.str, cval.len)) {
+		if (!F_ISSET(txn, WT_TXN_RUNNING))
```

Comment:
I think this conditional should be moved after the STAT increment. Particularly if/when the other timestamps are added here, it is a general conditional that would apply to all of them.